storage/pruning/triePersistersTracker.go

```diff
@@ -0,0 +1,79 @@
+package pruning
+
+import (
+	"bytes"
+
+	"github.com/ElrondNetwork/elrond-go/common"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type triePersistersTracker struct {
+	oldestEpochKeep      int64
+	oldestEpochActive    int64
+	numDbsMarkedAsActive int
+	numDbsMarkedAsSynced int
+}
+
+// NewTriePersisterTracker creates a new instance of triePersistersTracker
+func NewTriePersisterTracker(args *StorerArgs) *triePersistersTracker {
+	oldestEpochActive, oldestEpochKeep := computeOldestEpochActiveAndToKeep(args)
+
+	return &triePersistersTracker{
+		oldestEpochKeep:      oldestEpochKeep,
+		oldestEpochActive:    oldestEpochActive,
+		numDbsMarkedAsActive: 0,
+		numDbsMarkedAsSynced: 0,
+	}
+}
+
+// HasInitializedEnoughPersisters returns true if enough persisters have been initialized
+func (tpi *triePersistersTracker) HasInitializedEnoughPersisters(epoch int64) bool {
+	shouldKeepEpoch := epoch >= tpi.oldestEpochKeep
+	if shouldKeepEpoch {
+		return false
+	}
+
+	storageHasBeenSynced := tpi.numDbsMarkedAsActive >= 1 && tpi.numDbsMarkedAsSynced >= 1
+	if storageHasBeenSynced {
+		return true
+	}
+
+	return tpi.hasActiveDbsNecessary()
+}
+
+func (tpi *triePersistersTracker) hasActiveDbsNecessary() bool {
+	return tpi.numDbsMarkedAsActive >= minNumOfActiveDBsNecessary
+}
+
+// ShouldClosePersister returns true if the given persister needs to be closed
+func (tpi *triePersistersTracker) ShouldClosePersister(p storage.Persister, epoch int64) bool {
+	shouldClosePersister := epoch < tpi.oldestEpochActive && tpi.hasActiveDbsNecessary()
+
+	if isDbMarkedAsActive(p) {
+		tpi.numDbsMarkedAsActive++
```

Comment:
I think we need to protect with a mutex numDbsMarkedAsActive and numDbsMarkedAsSynced

Reply:
What is the scenario in which you see a possible concurrency issue? The persisterTracker is initialized on the PruningStorer constructor, and used only on that constructor. There are no goroutines used on the constructor. Or are you thinking to use mutex protection from future misuses of this component?

Reply:
ok, if that is the only use case, we can drop the usage of the mutex but we should state in the comments of the component/functions that this is not a concurrent safe component/function.

Reply:
Done.